documentation/optee_design.md

```diff
@@ -404,180 +239,165 @@ thread_svc_regs` in case the TA is executed in AArch64.
+a capable TEE solution in SRAM. The pager provides a solution to this by demand
+paging readonly parts of OP-TEE using virtual memory.
 
 ## Secure memory
-TrustZone protected SRAM is generally considered more secure than
-TrustZone protected DRAM as there's usually more attack vectors on DRAM.
-The attack vectors are hardware dependent and can be different for
-different platforms.
+TrustZone protected SRAM is generally considered more secure than TrustZone
+protected DRAM as there's usually more attack vectors on DRAM. The attack
+vectors are hardware dependent and can be different for different platforms.
 
 ## Backing store
-TrustZone protected DRAM or in some cases non-secure DRAM is used as
-backing store. The data in the backing store is integrity protected with
-one hash (SHA-256) per page (4KiB). Only readonly pages are kept in backing
-store for performance reasons as readwrite data would need to be encrypted
-too. Readonly pages doesn't need to be encrypted since the OP-TEE binary
-itself is not encrypted.
+TrustZone protected DRAM or in some cases non-secure DRAM is used as backing
+store. The data in the backing store is integrity protected with one hash
+(SHA-256) per page (4KiB). Only readonly pages are kept in backing store for
+performance reasons as readwrite data would need to be encrypted too. Readonly
+pages doesn't need to be encrypted since the OP-TEE binary itself is not
+encrypted.
 
 ## Partitioning of memory
 The code that handles demand paging must always be available as it would
 otherwise lead to deadlock. The virtual memory is partitioned as:
 
 ```
   Type      Sections
-+---------+-----------------+
-|         | text            |
-|         | rodata          |
-|         | data            |
-| unpaged | bss             |
-|         | heap1           |
-|         | nozi            |
-|         | heap2           |
-+---------+-----------------+
-| init /  | text_init       |
-| paged   | rodata_init     |
-+---------+-----------------+
-| paged   | text_pageable   |
-|         | rodata_pageable |
-+---------+-----------------+
-| demand  |                 |
-| alloc   |                 |
-+---------+-----------------+
++--------------+-----------------+
+|              | text            |
+|              | rodata          |
+|              | data            |
+| unpaged      | bss             |
+|              | heap1           |
+|              | nozi            |
+|              | heap2           |
++--------------+-----------------+
+| init / paged | text_init       |
+|              | rodata_init     |
++------------- +-----------------+
+| paged        | text_pageable   |
+|              | rodata_pageable |
++--------------+-----------------+
+| demand alloc |                 |
+|              |                 |
++--------------+-----------------+
 ```
-Where "nozi" stands for "not zero initialized", this section contains small
+Where "`nozi`" stands for "not zero initialized", this section contains small
```

Comment:
s/small stacks/entry stacks/
Add "thread stack (when TEE pager is disable)".
Add "TEE pager cached translation table (in case pager is enable and LPAE MMU is used)".